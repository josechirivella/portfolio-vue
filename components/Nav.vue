<template>
  <nav class="w-full sticky top-0">
    <div class="container mx-auto p-4 lg:flex">
      <div
        class="w-full relative flex justify-between items-center lg:w-auto lg:static lg:block lg:justify-start"
      >
        <NuxtLink
          class="text-lg font-bold leading-relaxed inline-block mr-4 py-2 whitespace-nowrap text-white"
          to="/"
        >
          Jose Chirivella
        </NuxtLink>
        <button
          class="text-white cursor-pointer text-xl leading-none px-3 py-1 rounded bg-transparent block lg:hidden outline-none focus:outline-none"
          type="button"
          @click="toggleNavbar()"
        >
          <Icon name="ci:hamburger-md" color="white" />
        </button>
      </div>
      <div
        :class="{ hidden: !showMenu, flex: showMenu }"
        class="lg:flex lg:flex-grow items-center"
      >
        <ul class="flex flex-col lg:flex-row list-none lg:ml-auto">
          <li v-for="item in navItems" :key="item.name" class="nav-item">
            <NuxtLink
              class="px-3 py-2 flex items-center text-lg font-bold leading-snug"
              :to="item.link"
              :target="item.target"
            >
              <i
                class="text-lg leading-lg text-white opacity-75"
              /><span class="ml-2 text-gray-400  hover:text-white">{{ item.name }}</span>
            </NuxtLink>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</template>

<script setup lang="ts">
import { ref } from "vue";
interface INavItems {
  name: string;
  link: string;
  target?: string;
}
const showMenu = ref(false);
const navItems: Array<INavItems> = [
  {
    name: "Home",
    link: "/",
  },
  {
    name: "About",
    link: "/about",
  },
  {
    name: "Blog",
    link: "/blog",
  },
  {
    name: "Resume",
    link: "https://storage.googleapis.com/portfolio-assets/resume/ResumeJose.pdf",
    target: "_blank",
  },
];
function toggleNavbar() {
  showMenu.value = !showMenu.value;
}
</script>

<style lang="scss">
.nav-item {
  .router-link-active span {
    @apply text-white;
  }
}
</style>
