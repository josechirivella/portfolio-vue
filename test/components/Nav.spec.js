import { beforeAll, describe, expect, test, vi } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Nav from '~/components/Nav.vue';

// Mock the useRoute composable
const mockRoute = {
  name: 'index',
};

vi.mock('vue-router', () => ({
  useRoute: vi.fn(() => mockRoute),
}));

describe('Nav', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallowMount(Nav, {
      global: {
        stubs: {
          NuxtLink: true,
          Icon: true,
        },
      },
    });
  });

  test('should render component', () => {
    expect(wrapper.vm).toBeTruthy();
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  describe('Navigation items', () => {
    test('should have the correct number of navigation items', () => {
      expect(wrapper.vm.navItems.length).toBe(4);
    });

    test('should have the correct navigation item names', () => {
      const navItemNames = wrapper.vm.navItems.map((item) => item.name);
      expect(navItemNames).toContain('Home');
      expect(navItemNames).toContain('About');
      expect(navItemNames).toContain('Blog');
      expect(navItemNames).toContain('Resume');
    });
  });

  describe('Toggle navbar', () => {
    test('toggleNavbar should toggle showMenu value', async () => {
      // Initial state
      expect(wrapper.vm.showMenu).toBe(false);

      // Toggle
      await wrapper.vm.toggleNavbar();
      expect(wrapper.vm.showMenu).toBe(true);

      // Toggle again
      await wrapper.vm.toggleNavbar();
      expect(wrapper.vm.showMenu).toBe(false);
    });
  });

  describe('inBlog function', () => {
    test('should return false when route name is not blog-slug', () => {
      mockRoute.name = 'index';
      expect(wrapper.vm.inBlog()).toBe(false);
    });

    test('should return true when route name is blog-slug', () => {
      mockRoute.name = 'blog-slug';
      expect(wrapper.vm.inBlog()).toBe(true);
    });
  });
});
