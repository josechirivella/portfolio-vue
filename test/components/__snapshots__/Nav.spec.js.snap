// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Nav > should render component 1`] = `
<nav
  class="w-full sticky top-0"
>
  <div
    class="container mx-auto p-4 lg:flex"
  >
    <div
      class="w-full relative flex justify-between items-center lg:w-auto lg:static lg:block lg:justify-start"
    >
      <nuxt-link-stub
        class="text-lg font-bold leading-relaxed inline-block mr-4 py-2 whitespace-nowrap"
        to="/"
      />
      <button
        class="cursor-pointer text-xl leading-none px-3 py-1 rounded bg-transparent block lg:hidden outline-none focus:outline-none"
        type="button"
      >
        <nuxt-icon-stub
          color="white"
          name="ci:hamburger-md"
        />
      </button>
    </div>
    <div
      class="hidden lg:flex lg:flex-grow items-center"
    >
      <ul
        class="flex flex-col lg:flex-row list-none lg:ml-auto"
      >
        
        <li
          class="nav-item"
        >
          <nuxt-link-stub
            class="px-3 py-2 flex items-center text-lg font-bold leading-snug"
            to="/"
          />
        </li>
        <li
          class="nav-item"
        >
          <nuxt-link-stub
            class="px-3 py-2 flex items-center text-lg font-bold leading-snug"
            to="/about"
          />
        </li>
        <li
          class="nav-item"
        >
          <nuxt-link-stub
            class="px-3 py-2 flex items-center text-lg font-bold leading-snug"
            to="/blog"
          />
        </li>
        <li
          class="nav-item"
        >
          <nuxt-link-stub
            class="px-3 py-2 flex items-center text-lg font-bold leading-snug"
            target="_blank"
            to="https://storage.googleapis.com/portfolio-assets/resume/ResumeJose.pdf"
          />
        </li>
        
      </ul>
    </div>
  </div>
</nav>
`;
